test/source/tests/tooling/browser-recipe.ts

```diff
@@ -46,12 +46,8 @@ export class BrowserRecipe {
     const settingsPage = await browser.newExtensionSettingsPage(t, acctEmail);
     const oauthPopup = await browser.newPageTriggeredBy(t, () => settingsPage.waitAndClick('@action-connect-to-gmail'));
     await OauthPageRecipe.google(t, oauthPopup, acctEmail, 'approve');
-    if (checkForConfiguredIdPOAuth)
-      await settingsPage.waitAndRespondToModal(
-        'warning',
-        'confirm',
-        'Custom IdP is configured on this domain, but it is not supported on browser extension yet.'
-      );
+    if (checkForConfiguredIdPOAuth) await settingsPage.page.waitForNavigation({ waitUntil: 'networkidle2' });
```

Comment:
I would hope ideally here it would explicitly write OauthPageRecipe.customIdp(...) under the .google() line

Reply:
You're absolutely right. I also thought about this. The custom IDP popup appears automatically after the user completes Google authentication, without any user action. The challenge is getting the custom IDP popup instance since there's no user interaction. Need to think about a solution for this.

Reply:
If it makes it easier for you, you could do `OauthPageRecipe.googleFollowedByCustomIdp` so you're still reusing the instance. My goal is that usually it only does `OauthPageRecipe.google()` and in a few cases where we test for this, we do it explicitly

Reply:
Sorry, I don't quite get it.
Do you mean I should put the `OauthPageRecipe.google()` call and the `await settingsPage.page.waitForNavigation({ waitUntil: 'networkidle2' });` logic into `OauthPageRecipe.googleFollowedByCustomIdp` and then call that when `checkForConfiguredIdPOAuth` is true?

Reply:
What you mentioned is correct.

For testing, the custom IDP is not an actual IDP; it is just a mock. When the Google IDP mock is finished, it opens the mock `/o/oauth2/auth `URL and then automatically redirects to the success page (https://google.localhost:${port}/robots.txt). This page is then used to retrieve the auth code and get the custom IDP ID token.

https://github.com/FlowCrypt/flowcrypt-browser/blob/3f10b9ef5595428a3d34d07b6f728e6109bb48e8/test/source/mock/google/google-endpoints.ts#L137

> - user clicks to log in
> - Google IdP pops up, user interacts with Google IdP, Google IdP goes away
> - custom IdP pops up, user interacts with custom IdP, custom IdP goes away
> - success
> -

Reply:
FYI, I set `customIDPClientId` to the actual custom IDP client ID, but it can be any string, like `custom-test-idp-client-id`, and it doesn't matter at all.

`public static customIDPClientId = 'ZFfdspDb9Oz1vnFdA4NrCyNaA9N3jYDl';`

Reply:
For the Google IDP mock, after the user taps the `Sign in with Google` button, it opens the `/o/oauth2/auth` URL (without `proceed=true`). Therefore, we need to redirect the user to `/o/oauth2/auth?proceed=true` at `oauth-page-recipe.ts#L38`.

However, for the custom IDP mock, there's no check to see if `proceed=true`. Consequently, after the Google mock IDP is finished and the custom IDP mock window opens, it automatically redirects to `robots.txt` which is then used to get the authorization code and ID token, and the popup closes automatically.

It's a good idea to check if `proceed=true` in the custom IDP mock as well. However, as I mentioned, it's challenging to get a reference to the custom mock IDP window.

Let me know if this makes sense.

https://github.com/FlowCrypt/flowcrypt-browser/blob/8770025a965a3ad13cd8fa81d7c3f92fe8117499/test/source/tests/page-recipe/oauth-page-recipe.ts#L38


https://github.com/FlowCrypt/flowcrypt-browser/blob/3f10b9ef5595428a3d34d07b6f728e6109bb48e8/test/source/mock/google/google-endpoints.ts#L132

https://github.com/FlowCrypt/flowcrypt-browser/blob/3f10b9ef5595428a3d34d07b6f728e6109bb48e8/test/source/mock/google/google-endpoints.ts#L137

Reply:
Okay, i get it now. 

Actually, from user perspective, I think there should probably be a button anyway. So the second the window doesn't show up on its own. Instead an alert (for simplicity?) shows up "Please also log in with company credentials". and a "Log in" button

Then you can click that in ".customIdp()" in tests, and also wait for network on that method. 

Then when we expect custom login, you do ".gmail()" followed by ".customIdp()" and it's clear it's tested for. 

also in this new method, please copy what you said here about the lack of user interaction, as a comment. So it's clear why so different from .gmail

thanks!

Reply:
That sounds great.
Thank you for your clarification.

Reply:
> That sounds great.
> Thank you for your clarification.

In this case, `checkForConfiguredIdPOAuth` should be called `expectCustomIdp`